### ISSUE TITLE
Unhandled exception when exiting app on macOS

### ISSUE THREAD

**Issue by filipnavara** [AvaloniaUI/Avalonia#12459]
**Describe the bug**
An empty app on macOS crashes with an exception in the native code on Cmd+Q quit command. This in turn triggers a [debugger deadlock](https://github.com/dotnet/runtime/issues/90079) too.

**To Reproduce**

1. Create empty app (https://github.com/filipnavara/emptyavalonia)
2. Run it under VS Code debugger on macOS
3. Close the app with Cmd+Q
4. App exit hangs

**Expected behavior**
App exits cleanly.

**Screenshots**

**Desktop (please complete the following information):**

- OS: MacOS Sonoma beta 5
- Version 11

**Additional context**
Native stack trace:
```
    +   2064 dyld4::LibSystemHelpers::exit(int) const  (in libdyld.dylib) + 20  [0x181f8b1d0]
    +     2064 exit  (in libsystem_c.dylib) + 44  [0x181e32e34]
    +       2064 __cxa_finalize_ranges  (in libsystem_c.dylib) + 476  [0x181e330ac]
    +         2064 ComPtr<IAvnDispatcher>::~ComPtr()  (in libAvaloniaNative.dylib) + 36  [0x105c3a240]
    +           2064 ???  (in <unknown binary>)  [0x281155810]
    +             2064 PAL_DispatchExceptionWrapper  (in libcoreclr.dylib) + 16  [0x101267f74]
    +               2064 PAL_DispatchException  (in libcoreclr.dylib) + 136  [0x1012682e4]
    +                 2064 SEHProcessException(PAL_SEHException*)  (in libcoreclr.dylib) + 348  [0x101233b7c]
```

**Comment by MrJul** [comment]
This is 11.0.0 right? The cause should be fixed in 11.0.1, see #12248 for the fix.
Thanks for confirming the debugger deadlock!

**Comment by filipnavara** [comment]
Correct, it was 11.0.0. Thanks for fix!
